src/shared/callbacks.rs

```diff
@@ -96,39 +96,78 @@ impl Callback {
             Err(_e) => Err("Callback failed".to_string()),
         }
     }
+
+    pub fn call1(&self, ss: String) -> Result<Promise, String> {
+        match self.f.call1(&JsValue::NULL, &JsValue::from_str(&ss)) {
+            Ok(result) => Ok(Promise::from(result)),
+            Err(_e) => Err("Callback failed".to_string()),
+        }
+    }
 }
 
 unsafe impl<'a> Send for Callback {}
 unsafe impl<'a> Sync for Callback {}
 
+pub enum CallbackType {
+    Buckets,
+    Measurements { bucket: String },
+}
+
 #[derive(Default, Clone)]
 pub struct Callbacks {
     pub buckets: Option<Callback>,
+    pub measurements: Option<Callback>,
 }
 
 impl Callbacks {
-    pub fn register_buckets_callback(&mut self, f: Function) {
-        self.buckets = Some(Callback::new(f));
+    pub fn register_callback(
```

Comment:
I think its cleaner to have a register_measurements_callback function similar to how buckets waw